src/utils/https_proxy_utils/src/https_proxy_utils.c

```diff
@@ -53,109 +121,85 @@ bool InitializeProxyOptions(HTTP_PROXY_OPTIONS* proxyOptions)
-    }
+    unescape_data_string(proxy_copy);
 
-    // Get host address.
-    uc = curl_url_get(curlHandle, CURLUPART_HOST, &host, 0);
-    if (uc != CURLE_OK)
-    {
-        Log_Error("Bad hostname. uc:%d", uc);
-        goto done;
-    }
+    start = proxy_copy;
 
-    if (mallocAndStrcpy_s((char**)&(proxyOptions->host_address), host) != 0)
+    end = strstr(start, "://");
+    if (end == NULL)
     {
-        Log_Error("Failed to copy host address.");
+        // Can't find "://"
         goto done;
     }
 
-    uc = curl_url_get(curlHandle, CURLUPART_PORT, &port, 0);
-    if (uc != CURLE_OK)
-    {
-        Log_Info("No ports. uc:%d", uc);
-    }
-    else
-    {
-        char* endptr;
-        errno = 0; /* To distinguish success/failure after call */
-        proxyOptions->port = strtol(port, &endptr, 10);
-        if (errno != 0 || endptr == port)
-        {
-            Log_Error("Invalid port number.");
-            goto done;
-        }
-    }
+    end += 3; // past ""://""
 
-    uc = curl_url_get(curlHandle, CURLUPART_USER, &user, 0);
-    if (uc != CURLE_OK)
-    {
-        Log_Info("No username. uc:%d", uc);
-    }
-    else
-    {
-        if (mallocAndStrcpy_s((char**)&(proxyOptions->username), user) != 0)
-        {
-            goto done;
-        }
-    }
+    start = end;
 
-    uc = curl_url_get(curlHandle, CURLUPART_PASSWORD, &password, 0);
-    if (uc != CURLE_OK)
+    end = strchr(start, '@');
+    if (end != NULL)
     {
-        Log_Info("No password. uc:%d", uc);
-    }
-    else
-    {
-        if (mallocAndStrcpy_s((char**)&(proxyOptions->password), password) != 0)
+        // Left of '@' is username[:password]
+        *end = '\0';
+        char* after_at = end + 1; // after '@'
+
+        end = strchr(start, ':');
+        if (end != NULL)
         {
-            Log_Error("Failed to copy password");
-            goto done;
+            *end = '\0';
+
+            // Check for no username
+            if (*start != '\0')
+            {
+                proxyOptions->username = strdupe(start);
+                if (proxyOptions->username == NULL)
+                {
+                    goto done;
+                }
+            }
+
+            start = end + 1;
+
+            // Check for no password
+            if (*start != '\0')
+            {
+                proxyOptions->password = strdupe(start);
+                if (proxyOptions->password == NULL)
+                {
+                    goto done;
+                }
+            }
         }
-    }
-
-    success = true;
 
-done:
-
-    if (password != NULL)
-    {
-        curl_free(password);
+        start = after_at;
     }
 
-    if (user != NULL)
+    end = strchr(start, ':');
+    if (end != NULL)
     {
-        curl_free(user);
-    }
+        *end = '\0';
 
-    if (port != NULL)
-    {
-        curl_free(port);
+        proxyOptions->port = atoi(end + 1);
     }
 
-    if (host != NULL)
+    proxyOptions->host_address = strdupe(start);
```

Comment:
If we do not want to preclude customers who use international domain names who have built libcurl with IDN support, then need to do convert percent-decoded that correspond to UTF-8 bytes sequences (1 - 3 characters) as per [RFC3986](https://www.rfc-editor.org/rfc/rfc3986).

For example "%E3%82%A2" should become the KATAKANA LETTER A
and "%C3%80" should become LATIN CAPITAL LETTER A WITH GRAVE.

on cmdline, curl built with IDN support can download this: `curl http://www.malmö.se/file`

Here is Curl_idnconvert_hostname function to do IDN conversion of hostname:
https://github.com/curl/curl/blob/eb559c8056916f1e71570ac058d599347bebc2b4/lib/url.c#L1574

## Excerpt from RFC3986
```
"The reg-name syntax allows percent-encoded octets in order to
   represent non-ASCII registered names in a uniform way that is
   independent of the underlying name resolution technology.  Non-ASCII
   characters must first be encoded according to UTF-8 [[STD63](https://www.rfc-editor.org/rfc/rfc3986#ref-STD63)], and then
   each octet of the corresponding UTF-8 sequence must be percent-
   encoded to be represented as URI characters.  URI producing
   applications must not use percent-encoding in host unless it is used
   to represent a UTF-8 character sequence."
```

## libcurl can handle international domain names
https://curl.se/libcurl/c/CURLOPT_URL.html has this to say:

### Encoding
```
"The string pointed to in the [CURLOPT_URL](https://curl.se/libcurl/c/CURLOPT_URL.html) argument 
is generally expected to be a sequence of characters using an ASCII compatible encoding.

If libcurl is built with IDN support, the server name part of the URL can use an "international name" by 
using the current encoding (according to locale) or UTF-8 (when winidn is used; or a Windows Unicode 
build using libidn2).

If libcurl is built without IDN support, the server name is used exactly as specified when passed to the
 name resolver functions."
```